File: bloc_de_ingresos_mensuales.py
opcion = 0
i = 0
contador = 0
posicion = 0
encontrado = 0
ingreso = 0.0
total = 0.0
descripcion = ""
ingresos = []
descripciones = []
#-----------------------------------------------#
#--|menu_principal_bloc_de_ingresos_mensuales|--#
#-----------------------------------------------#
while opcion != 6:
    print("\n==============================")
    print("💰 BLOQUE DE INGRESOS MENSUALES")
    print("==============================")
    print("1. Agregar ingreso")
    print("2. Ver todos los ingresos")
    print("3. Buscar ingreso")
    print("4. Calcular total mensual")
    print("5. Eliminar ingreso")
    print("6. Salir")
    opcion = int(input("Seleccione una opcion: "))
    #---------------------#
    #--|agregar_ingreso|--#
    #---------------------#
    if opcion == 1:
        descripcion = input("Ingrese descripcion del ingreso: ")
        ingreso = float(input("Ingrese monto del ingreso: "))
        descripciones.append(descripcion)
        ingresos.append(ingreso)
        contador += 1
        print("✔ Ingreso agregado correctamente")
    #----------------------------#
    #--|ver_todos_los_ingresos|--#
    #----------------------------#
    elif opcion == 2:
        if contador == 0:
            print("No hay ingresos registrados")
        else:
            for i in range(contador):
                print(i + 1, ".", descripciones[i], "-", "$", ingresos[i])
    #--------------------#
    #--|buscar_ingreso|--#
    #--------------------#
    elif opcion == 3:
        descripcion = input("Ingrese descripcion a buscar: ")
        encontrado = 0
        for i in range(contador):
            if descripciones[i] == descripcion:
                print("✔ Encontrado:", descripciones[i], "-", "$", ingresos[i])
                encontrado = 1
        if encontrado == 0:
            print("❌ No se encontro el ingreso")
    #------------------#
    #--|total_mesual|--#
    #------------------#
    elif opcion == 4:
        total = 0
        for i in range(contador):
            total += ingresos[i]
        print("🧮 Total mensual de ingresos: $", total)
    #----------------------#
    #--|eliminar_ingreso|--#
    #----------------------#
    elif opcion == 5:
        posicion = int(input("Ingrese posicion del ingreso a eliminar (1 - {}): ".format(contador)))
        if posicion >= 1 and posicion <= contador:
            ingresos.pop(posicion - 1)
            descripciones.pop(posicion - 1)
            contador -= 1
            print("🗑️ Ingreso eliminado correctamente")
        else:
            print("❌ Posicion invalida")
    #------------------------------#
    #--|salir_del_menu_principal|--#
    #------------------------------#
    elif opcion == 6:
        print("🚪 Saliendo del sistema...")
    else:
        print("❌ Opcion invalida")

File: bloc_de_ingresos_mensuales.java
import java.util.ArrayList;
import java.util.Scanner;
public class bloc_de_ingresos_mensuales {
    public static void main(String[] args) {
        int opcion = 0;
        int i = 0;
        int contador = 0;
        int posicion = 0;
        int encontrado = 0;
        double ingreso = 0.0;
        double total = 0.0;
        String descripcion = "";
        ArrayList<Double> ingresos = new ArrayList<>();
        ArrayList<String> descripciones = new ArrayList<>();
        Scanner sc = new Scanner(System.in);
        //-----------------------------------------------//
        //--|menu_principal_bloc_de_ingresos_mensuales|--//
        //-----------------------------------------------//
        while (opcion != 6) {
            System.out.println("\n==============================");
            System.out.println("💰 BLOQUE DE INGRESOS MENSUALES");
            System.out.println("==============================");
            System.out.println("1. Agregar ingreso");
            System.out.println("2. Ver todos los ingresos");
            System.out.println("3. Buscar ingreso");
            System.out.println("4. Calcular total mensual");
            System.out.println("5. Eliminar ingreso");
            System.out.println("6. Salir");
            System.out.print("Seleccione una opcion: ");
            opcion = sc.nextInt();
            sc.nextLine();
            //---------------------//
            //--|agregar_ingreso|--//
            //---------------------//
            if (opcion == 1) {
                System.out.print("Ingrese descripcion del ingreso: ");
                descripcion = sc.nextLine();
                System.out.print("Ingrese monto del ingreso: ");
                ingreso = sc.nextDouble();
                sc.nextLine();
                descripciones.add(descripcion);
                ingresos.add(ingreso);
                contador++;
                System.out.println("✔ Ingreso agregado correctamente");
            }
            //----------------------------//
            //--|ver_todos_los_ingresos|--//
            //----------------------------//
            else if (opcion == 2) {
                if (contador == 0) {
                    System.out.println("No hay ingresos registrados");
                } else {
                    for (i = 0; i < contador; i++) {
                        System.out.println((i + 1) + ". " + descripciones.get(i) + " - $" + ingresos.get(i));
                    }
                }
            }
            //--------------------//
            //--|buscar_ingreso|--//
            //--------------------//
            else if (opcion == 3) {
                System.out.print("Ingrese descripcion a buscar: ");
                descripcion = sc.nextLine();
                encontrado = 0;
                for (i = 0; i < contador; i++) {
                    if (descripciones.get(i).equalsIgnoreCase(descripcion)) {
                        System.out.println("✔ Encontrado: " + descripciones.get(i) + " - $" + ingresos.get(i));
                        encontrado = 1;
                    }
                }
                if (encontrado == 0) {
                    System.out.println("❌ No se encontro el ingreso");
                }
            }
            //------------------//
            //--|total_mesual|--//
            //------------------//
            else if (opcion == 4) {
                total = 0;
                for (i = 0; i < contador; i++) {
                    total += ingresos.get(i);
                }
                System.out.println("🧮 Total mensual de ingresos: $" + total);
            }
            //----------------------//
            //--|eliminar_ingreso|--//
            //----------------------//
            else if (opcion == 5) {
                System.out.print("Ingrese posicion del ingreso a eliminar (1 - " + contador + "): ");
                posicion = sc.nextInt();
                sc.nextLine();
                if (posicion >= 1 && posicion <= contador) {
                    ingresos.remove(posicion - 1);
                    descripciones.remove(posicion - 1);
                    contador--;
                    System.out.println("🗑️ Ingreso eliminado correctamente");
                } else {
                    System.out.println("❌ Posicion invalida");
                }
            }
            //------------------------------//
            //--|salir_del_menu_principal|--//
            //------------------------------//
            else if (opcion == 6) {
                System.out.println("🚪 Saliendo del sistema...");
            }
            else {
                System.out.println("❌ Opcion invalida");
            }
        }
        sc.close();
    }
}